docs/telemetry.rst

```diff
@@ -165,3 +166,25 @@ Supported telemetry parameters:
 
 * ``searchable-snapshots-stats-indices`` (default: None): A string with the index/index pattern, or list of indices/index patterns that searchable snapshots stats should additionally be collected from. If unset, only cluster level stats will be collected.
 * ``searchable-snapshots-stats-sample-interval`` (default 1): A positive number greater than zero denoting the sampling interval in seconds.
+
+shard-stats
+--------------
+
+The shard-stats telemetry device regularly calls the `cluster node-stats API with level=shard parameter <https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-nodes-stats.html>`_ and records one metrics document per shard.
+
+Example of the recorded document::
+
+   {
+     "name": "shard-stats",
+     "shard-id": "0",
+     "index": "geonames",
+     "primary": True,
+     "docs": 1000,
+     "store": 212027,
+     "segments_count": 8,
```

Comment:
We mix hyphens (see `shard-id`) and underscores here but it would be nicer to read if this is consistent. Can you please use hyphens everywhere?